### PR TITLE
ORC-1285: [Java] Refactor selected vector support

### PR DESCRIPTION
**Pull request by deshanxiao**
### What changes were proposed in this pull request?
This pr is aimed to refactor the support of selected vector.

### Why are the changes needed?
reduce code duplication

### How was this patch tested?
UT(TestSelectedVector)

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/WriterImpl.java [hunk 1]
```diff
@@ -1018,4 +1005,20 @@ private WriterEncryptionVariant[] setupEncryption(
...
     return this.treeWriter.estimateMemory();
   }
+
+  /**
+   * Get next chunk size if row selected enable.
+   * We will find the longest size of chunk that will be less than maxSize.
+   * @param posn the position where we are
+   * @param maxSize the largest chunk size in this selection
+   */
+  private int nextSelectedRowSize(int posn, int maxSize, int[] selected) {
```
Comment (wgtmac) [inline java/core/src/java/org/apache/orc/impl/WriterImpl.java:1015]: Good refactoring. The only concern is that before the change we know posn and (posn + len) will never exceed selected.length. The new method has potential misuse if someone else has passed wrong maxSize when calling it.

Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/WriterImpl.java:1015]: Good catching! Will add an assertion here.

**Comment by deshanxiao** [comment]
Also CC @dongjoon-hyun @guiyanakuang Thanks~

**Comment by dongjoon-hyun** [comment]
cc @pavibhai too

**Pull request closed without merging**